File: HackerRank/Time_Conversion.py


import os
import sys



def timeConversion(s):
    x= slice(8,10)
    ses = s[x]
    hr = s[0:2]
    re = s[0:8]
    mo = "00"

    if ses=="PM":
        if hr=="12":
            return re
        else:
            nwhr = str(int(hr)+12)
            return (nwhr+s[2:8])
    elif hr=="12" and ses=="AM":
        return mo+s[2:8]
    else:
        return re
     
    

s = input()

result = timeConversion(s)

print(result + '\n')

    

File: HackerRank/timeConversion.rb
def timeConversion(s)
    sp = s[s.length-2, s.length-1]
    s.slice!(sp)
    s = s.split(':').map{ |x| x.to_i }

    s[0] = (sp == 'AM') ? s[0] == 12 ? 0 : s[0] : s[0] == 12 ? 12 : s[0] + 12
    
    return s.map{ |x| '%02d' % x }.join(':')
end

s = gets.to_s

result = timeConversion s

puts result